soc/intel/intel_adsp/tools/cavstool.py

```diff
@@ -188,17 +191,34 @@ def reset(self):
         self.debug()
         log.info(f"Reset stream {self.stream_id}")
 
+def adsp_is_cavs():
+    return cavs15 or cavs18 or cavs15
+
+def adsp_is_ace():
+    return ace15 or ace20
+
+def adsp_mem_window_config():
```

Comment:
I guess this is a runtime function vs. just setting the variables because they're global and you haven't enumerated the PCI bus yet at their definition?  Seems clumsy, but then doing bigger reorganization in a patch like this is probably a bad idea anyway.  Something to look at in the longer term, maybe.

Reply:
Yeah, I tried to keep the design as such and not add any new abstractions (maybe good in light of @teburd 's comment on earlier try). The existing code sets these global booleans based on the PCI DID, and I continued with this and added these helpers. And indeed, this is now a runtime check than depends on map_regs() results.